src/sql/workbench/services/connection/browser/connectionBrowseTab.ts

```diff
@@ -317,15 +343,21 @@ class IdentityProvider implements IIdentityProvider<TreeElement> {
 }
 
 class TreeModel {
+	private _savedConnectionNode: SavedConnectionNode | undefined;
 
 	constructor(
 		@IConnectionTreeService private readonly connectionTreeService: IConnectionTreeService,
 		@IInstantiationService private readonly instantiationService: IInstantiationService
 	) { }
 
 	getChildren(): TreeElement[] {
+		this._savedConnectionNode = this.instantiationService.createInstance(SavedConnectionNode);
```

Comment:
Why not just create this when the model is constructed?

Reply:
a new node will be created when the tree is refreshed, the property should return the newest 'Saved Connections' Node.